code/game/atoms.dm

```diff
@@ -83,6 +83,9 @@
 	if(opacity && isturf(loc))
 		var/turf/T = loc
 		T.has_opaque_atom = TRUE // No need to recalculate it in this case, it's guranteed to be on afterwards anyways.
+	
+	if(loc)
+		loc.InitializedOn(src) // Used for poolcontroller / pool to improve performance greatly. However it also open up path to other usage of observer pattern on turfs.

```

Comment:
Rather than having an entire atom proc just for this, why not just make the pool controller link itself to the turfs on poolcontroller/Initialize...?

Reply:
For the most part, this feature was added in so that when a decal to added to a pool turf, the pool turf can "listen to" the turf being added - And then add it to a list to be cleaned, which improves performance a lot.

However in the future this might be useful for other thing. For example, a closet with something created inside.